File: zig/src/indicators/core/pane.zig
/// Identifies the chart pane an indicator output is drawn on.
pub const Pane = enum(u8) {
    /// The primary price pane.
    price = 1,
    /// A dedicated sub-pane for this indicator.
    own = 2,
    /// Drawing on the parent indicator's pane.
    overlay_on_parent = 3,

    pub fn asStr(self: Pane) []const u8 {
        return switch (self) {
            .price => "price",
            .own => "own",
            .overlay_on_parent => "overlayOnParent",
        };
    }
};

File: rust/src/indicators/core/pane.rs
/// Identifies the chart pane an indicator output is drawn on.
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[repr(u8)]
pub enum Pane {
    /// The primary price pane.
    Price = 1,
    /// A dedicated sub-pane for this indicator.
    Own = 2,
    /// Drawing on the parent indicator's pane.
    OverlayOnParent = 3,
}

impl Pane {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::Price => "price",
            Self::Own => "own",
            Self::OverlayOnParent => "overlayOnParent",
        }
    }
}

impl std::fmt::Display for Pane {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_str(self.as_str())
    }
}
